e2e/nomostest/reset.go

```diff
@@ -478,34 +479,59 @@ func initialCommit(nt *NT, repoType gitproviders.RepoType, nn types.NamespacedNa
 	}
 }
 
-// TailReconcilerLogs starts tailing a reconciler's logs.
-// The logs are stored in memory until either the context is cancelled or the
-// kubectl command exits (usually because the container exited).
+// TailCSPodsAndReconcilerLogs starts tailing CS Pods and the reconciler's logs.
+// The pods and logs are stored in memory until either the context is cancelled
+// or the kubectl command exits (usually because the container exited).
 // This allows capturing logs even if the reconciler is deleted before the
 // test ends.
-// The logs will only be printed if the test has failed when the command exits.
-// Run in an goroutine to capture logs in the background while deleting RSyncs.
-func TailReconcilerLogs(ctx context.Context, nt *NT, reconcilerNN types.NamespacedName) {
-	out, err := nt.Shell.KubectlContext(ctx, "logs",
-		fmt.Sprintf("deployment/%s", reconcilerNN.Name),
-		"-n", reconcilerNN.Namespace,
-		"-c", reconcilermanager.Reconciler,
-		"-f")
-	nt.T.Logf("finished tailing logs for %s", reconcilerNN.String())
-	// Expect the logs to tail until the context is cancelled, or exit early if
-	// the reconciler container exited.
-	if err != nil && err.Error() != "signal: killed" {
-		// We're only using this for debugging, so don't trigger test failure.
-		nt.T.Logf("Failed to tail logs from reconciler %s: %v", reconcilerNN, err)
-	}
-	// When TailReconcilerLogs is called async, it's possible that the test fails
+// The pods and logs will only be printed if the test has failed when the
+// command exits.
+// Run in a goroutine to capture logs in the background while deleting RSyncs.
+func TailCSPodsAndReconcilerLogs(ctx context.Context, nt *NT, reconcilerNNs ...types.NamespacedName) {
```

Comment:
I don't think this will have the desired output since this is done sequentially. The pod get/describe is done immediately at the beginning of the function/goroutine, which I expect won't have the desired information (we probably want to describe after entering an error state). The log tailing is also being done sequentially, which I expect means we won't actually capture the logs for each reconciler. By the time the first iteration of the loop stops tailing logs for the first deployment, the other deployments/pods may have been deleted. To tail logs for each reconciler deployment/pod, I expect we would need a goroutine for each reconciler.

Reply:
I think the simplest solution for log tailing would probably be to just keep this function scoped to a single reconcilerNN

Reply:
That makes sense. So for N reconcilers, we'll need N+1 goroutines to get and describe the current pods, and then tail each reconciler log.

Reply:
Per offline discussion, we decided to print test logs before resetting by merging the two cleanup blocks into a single one.